### ISSUE TITLE
Confusing 'Module ... is used in several stanzas' error

### ISSUE THREAD

**Issue by jmid** [ocaml/dune#6598]
I was just bitten by this error which confused me, as it mentions a module name `Aaa` not occurring in the listed stanza.
I suspect people (even) less dune-experienced than me will be similarly confused.

## Expected Behavior

```
File "dune", line 6, characters 0-0:
Stanza is missing a "modules" field.

To fix this error, you must specify an explicit "modules" field in every
library, executable, and executables stanzas in this dune file. Note that
each module cannot appear in more than one "modules" field - it must belong
to a single library or executable.
```

## Actual Behavior

```
File "dune", line 1, characters 0-0:
Error: Module "Aaa" is used in several stanzas:
- dune:1
- dune:6
To fix this error, you must specify an explicit "modules" field in every
library, executable, and executables stanzas in this dune file. Note that
each module cannot appear in more than one "modules" field - it must belong
to a single library or executable.

```

## Reproduction

1. create a directory `repro`
2. `touch repro/dune-workspace`
3. create a file `repro/dune-project` with the content `(lang dune 2.9)`
4. create two files `repro/aaa.ml` and `repro/bbb.ml` with the content `exit 0`
5. create a file `repro/dune` with the following content:
   ```
    (executable
     (name aaa)
     (modules aaa)
    )
    
    (test
     (name bbb))
   ```
6. `cd repro`
7. `dune runtest`

## Specifications

- Version of `dune` (output of `dune --version`): 3.6.1
- Version of `ocaml` (output of `ocamlc --version`): 5.0.0~beta2
- Operating system (distribution and version): Linux Mint, running kernel 5.4.0-132-generic

**Comment by Alizter** [comment]
The `test` stanza is picking up both modules `Aaa` and `Bbb` which is why Dune is complaining. You need to add a `(modules)` field to the `test` stanza marking only `bbb`. The test stanza shares pretty much the same fields as the `executable` stanza. https://dune.readthedocs.io/en/stable/dune-files.html#tests

**Comment by jmid** [comment]
> The test stanza is picking up both modules `Aaa` and `Bbb` which is why Dune is complaining.

Yes, an error message explaining that would be helpful - and an improvement over the current one.
